### PR TITLE
DP-393 Progress bar for installers

### PR DESCRIPTION
**Pull request by tomonorman**
Needs testing, but heres the code!

### CODE REVIEW

File: installers/source/setup.sh [hunk 1]
```diff
@@ -84,18 +37,163 @@ echo_with_color() {
...
+print_centered "-" "-"
+print_centered "-" "-"
+print_centered "Welcome to DreamFactory!"
+print_centered "-" "-"
+print_centered "-" "-"
+print_centered "Thank you for choosing DreamFactory. Please select any additional installation options from the menu choices below (you may select more than one), or simply press enter to continue with the default installation (Nginx Webserver, latest version of DreamFactory).\n" 
+print_centered "-" "-"
+echo -e ""
+echo -e "[1] Install driver and PHP extensions for Oracle DB" 
+echo -e "[2] Install driver and PHP extensions for IBM DB2"
```
Comment (yaroslavmo) [inline installers/source/setup.sh:93]: In case I want Oracle and IBM DB2 to be installed, how to arrange that?

File: installers/source/setup.sh [hunk 2]
```diff
@@ -84,18 +37,163 @@ echo_with_color() {
...
+echo -e "[2] Install driver and PHP extensions for IBM DB2" 
+echo -e "[3] Install driver and PHP extensions for Cassandra DB" 
+echo -e "[4] Install Apache2 web server for DreamFactory (Instead of Nginx)" 
+echo -e "[5] Install MariaDB as the default system database for DreamFactory" 
+echo -e "[6] Install a specfic version of DreamFactory" 
+echo -e "[7] Run Installation in debug mode (logs will output to /tmp/dreamfactory_installer.log)\n"
+print_centered "-" "-"
+echo -e "E.g input 1,4 and press Enter to additionaly install Oracle Drivers and use Apache2 instead of Nginx" 
+print_centered "-" "-"
+echo_with_color magenta "Input additional installation options or leave blank for default installation and press Enter to continue"
```
Comment (yaroslavmo) [inline installers/source/setup.sh:102]: What are additional installation options?

**Review by yaroslavmo — COMMENTED** [review]
I like how you handled this ticket. The installer looks very neat now

File: installers/dfsetup.run [hunk 3]
```diff
@@ -0,0 +1,794 @@
...
+MD5="abe6548bcd423b19463a45d987f4e6a9"
+SHA="0000000000000000000000000000000000000000000000000000000000000000"
+SIGNATURE=""
+TMPROOT=${TMPDIR:=/tmp}
+USER_PWD="$PWD"
+export USER_PWD
+ARCHIVE_DIR=`dirname "$0"`
+export ARCHIVE_DIR
+
+label="DreamFactory 4.11.0"
```
[the remark this answers is not in the document]
Comment (tomonorman) [reply installers/dfsetup.run:20]: Nope not necessary, that label is built from the zipping process, can call it whatever. I'll change it up

File: installers/source/setup.sh (hunk 1 above)
Comment (tomonorman) [reply installers/source/setup.sh:93]: you would do `12` or `1,2` (or well any string as long as it contains a 1 and a 2) and press enter. I think the text needs to be made clear on the menu section. I'll have a think

File: installers/source/setup.sh (hunk 2 above)
Comment (tomonorman) [reply installers/source/setup.sh:102]: As above, I think the text needs rewording to make it a bit more user friendly

**Pull request merged**